bokehjs/src/lib/models/tools/inspectors/hover_tool.ts

```diff
@@ -207,22 +216,26 @@ export class HoverToolView extends InspectToolView {
     const {sx, sy} = ev
     if (!this.plot_view.frame.bbox.contains(sx, sy))
       this._clear()
-    else
+    else {
+      this._current_sxy = [sx, sy]
       this._inspect(sx, sy)
+    }
   }
 
   override _move_exit(): void {
+    this._current_sxy = null
     this._clear()
   }
 
   _inspect(sx: number, sy: number): void {
-    let geometry: PointGeometry | SpanGeometry
-    if (this.model.mode == "mouse")
-      geometry = {type: "point", sx, sy}
-    else {
-      const direction = this.model.mode == "vline" ? "h" : "v"
-      geometry = {type: "span", direction, sx, sy}
-    }
+    const geometry: PointGeometry | SpanGeometry = (() => {
```

Comment:
I personally prefer the original syntax which doesn't use a lambda function. I see that the lambda allows us to use `const` instead of `let`, but are there any other benefits of the lambda approach?

Reply:
The primary benefit is to reduce mutable `let` bindings to the minimum. It also allows to hide dependencies from being reachable in the later code (not really applicable here, because `direction` was already scoped). I don't like the syntax as well, but the benefits outweigh the ugliness. There's an initial proposal to add `do {}` expressions to JavaScript (see [here](https://github.com/tc39/proposal-do-expressions), which would improve this slightly, to:
```ts
    const geometry: PointGeometry | SpanGeometry = do {
      if (this.model.mode == "mouse")
        return {type: "point", sx, sy}
      else {
        const direction = this.model.mode == "vline" ? "h" : "v"
        return {type: "span", direction, sx, sy}
      }
    }
```
(if it wasn't for `{...}` syntax, `return` statements are supposed to be dropped as well in this context).

Reply:
Thanks for clarifying.

Reply:
One caveat though. Using IIFE in performance critical code and especially in tight loops (like in glyph data processing) can cause performance regressions, so I try to avoid that. Though this probably depends on the JS JIT in use, because I would expect a decent JIT to optimize IIFEs out in hot paths.